### ISSUE TITLE
Identify synergies with birdhouse project

### ISSUE THREAD

**Issue by huard** [pangeo-data/pangeo#220]
Hi all, 
I'm involved in the [birdhouse](https://github.com/bird-house) project whose objective is to offer climate services through OGC's Web Processing Services (WPS) standard. The philosophy, as with Pangeo, is to process data remotely rather than having to download it. One major difference is that we're not trying to provide a full programming environment, but only a library of key analytical processes that are often used. Scientists would run these processes remotely, get the output then use whatever environment they're familiar with to complete the analysis. We're slowly building an infrastructure where these processes can be chained in workflows. 

I think it would make sense for us to use the tools developed in Pangeo in our computational backend. We're using OCGIS, which as I understand could eventually support xarray. 

So basically I just wanted to say hi and let you know of our interest for your work.

**Comment by tomLandry** [comment]
Hello everyone. I work at Computer research institute of Montreal (CRIM) and I'm a close collaborator to @huard at Ouranos. My institution isn't hosting any climate scientist, but we do have in house several data scientists, remote sensing specialists, web developers, Big Data specialist, engineers, researchers and so on. We are also active in the birdhouse project, trying to expand its Earth observations capabilities, amongst other things.

I therefore reiterate David's notice of interest. Oh... and hi!

**Comment by jacobtomlinson** [comment]
Hi both of you! Welcome aboard!

I would be keen to know more. You mention that your project is `only a library` but then you mention you are `building an infrastructure`. Where does the library fit into the infrastructure?

**Comment by rabernat** [comment]
Welcome! This sounds like a great area for collaboration. So far we have mostly been focused on the interactive jupyter experience, but the same basic infrastructure should be able to form the basis of scalable services and APIs such as the one you described. WPS targets the GIS community, and we are still working on trying to define the relationship of Pangeo to the GIS world (see e.g. #216, #66) Most of us come from a different background and are not super familiar with GIS technologies.

**Comment by huard** [comment]
@jacobtomlinson I'll still try to answer it ; )
What I mean is that users only see the library (not in the .lib sense, but the collection sense) of processes that we provide. On the backend however, there is an architecture built to serve these processes, load balance them, deal with access permissions to datasets and services, etc.

**Comment by jacobtomlinson** [comment]
Ah ok, so it's more like an API which provides an OGC WPS service. And you have a library which allows you to use the service in your code?

**Comment by tomLandry** [comment]
For a more thorough view on Birdhouse, I post here a poster made by Carsten and al. from DKRZ. So this is part of the response you where looking for wrt librairies vs infrastructure, at the EU level. We also have a plan solidly in place for Canada infrastructure, but not public.
https://drive.google.com/file/d/1hCYd6TIad_AtigemqGlumilkcn1Zo9vH/view?usp=sharing

**Comment by huard** [comment]
Exactly, I've heard someone is working on WPS support for swagger, don't know the schedule for this though. 

Totally agree. That's why the client makes sense. Provides you with backed-in functionality in an environment that let's you experiment. 

Once things cool down on our end, we'll give Pangeo a spin and come back with more concrete ideas for collaboration.

**Comment by tomLandry** [comment]
I say we keep the issue open. We'll coordinate (remote) participation for the Pangeo developer meeting.
